libmscore/scorefile.cpp

```diff
@@ -413,8 +413,34 @@ bool MasterScore::saveFile()
             //
             // step 2
             // remove old backup file if exists
+            // remove the backup file in the same dir as score (the traditional place) if exists
             //
+            QString backupDirString = info.path() + QString(QDir::separator()) + ".backup";
+            QDir backupDir(backupDirString);
+            if (!backupDir.exists()) {
+                  dir.mkdir(".backup");
+#ifdef Q_OS_WIN
+                  QString backupDirNativePath = QDir::toNativeSeparators(backupDirString);
+#if (defined (_MSCVER) || defined (_MSC_VER))
+   #if (defined (UNICODE))
+                  SetFileAttributes((LPCTSTR)backupDirNativePath.unicode(), FILE_ATTRIBUTE_HIDDEN);
+   #else
+                  // Use byte-based Windows function
+                  SetFileAttributes((LPCTSTR)backupDirNativePath.toLocal8Bit(), FILE_ATTRIBUTE_HIDDEN);
+   #endif
+#else
+                  SetFileAttributes((LPCTSTR)backupDirNativePath.toLocal8Bit(), FILE_ATTRIBUTE_HIDDEN);
+#endif
+#endif
+                  }
             QString backupName = QString(".") + info.fileName() + QString(",");
+            if (backupDir.exists(backupName)) {
+                  if (!backupDir.remove(backupName)) {
+//                      if (!MScore::noGui)
+//                            QMessageBox::critical(0, QObject::tr("Save File"),
+//                               tr("Removing old backup file %1 failed").arg(backupName));
+                        }
+                  }
             if (dir.exists(backupName)) {
                   if (!dir.remove(backupName)) {
```

Comment:
Unless I miss something this removes the old backup file (which is good, for compatibility), but not the new one, and so on the next round the rename will fail due to target file existing!

Reply:
Why?

https://github.com/musescore/MuseScore/pull/5380/files#diff-daba2bf61c6ac304007b2fd76fb28285R438

removes the new backup file. Doesn't it?

Reply:
eh... yes ;-)
As I wrote "Unless I messsomething"... Seems I need new glasses